spec/components/chat_message_creator_spec.rb

```diff
@@ -208,6 +210,80 @@
       }.to change { user2.chat_mentions.count }.by(0)
     end
 
+    describe "group mentions" do
+      it "creates chat mentions for group mentions where the group is mentionable" do
+        expect {
+          DiscourseChat::ChatMessageCreator.create(
+            chat_channel: public_chat_channel,
+            user: user1,
+            content: "hello @#{admin_group.name}"
+          )
+        }.to change { admin1.chat_mentions.count }.by(1)
+          .and change {
+            admin2.chat_mentions.count
+          }.by(1)
+      end
+
+      it "doesn't mention users twice if they are direct mentioned and group mentioned" do
+        expect {
+          DiscourseChat::ChatMessageCreator.create(
+            chat_channel: public_chat_channel,
+            user: user1,
+            content: "hello @#{admin_group.name} @#{admin1.username} and @#{admin2.username}"
+          )
+        }.to change { admin1.chat_mentions.count }.by(1)
+          .and change {
+            admin2.chat_mentions.count
+          }.by(1)
+      end
+
+      it "creates chat mentions for group mentions and direct mentions" do
+        expect {
+          DiscourseChat::ChatMessageCreator.create(
+            chat_channel: public_chat_channel,
+            user: user1,
+            content: "hello @#{admin_group.name} @#{user2.username}"
+          )
+        }.to change { admin1.chat_mentions.count }.by(1)
+          .and change {
+            admin2.chat_mentions.count
+          }.by(1)
+          .and change {
+            user2.chat_mentions.count
+          }.by(1)
+      end
+
+      it "creates chat mentions for group mentions and direct mentions" do
+        expect {
+          DiscourseChat::ChatMessageCreator.create(
+            chat_channel: public_chat_channel,
+            user: user1,
+            content: "hello @#{admin_group.name} @#{user_group.name}"
+          )
+        }.to change { admin1.chat_mentions.count }.by(1)
```

Comment:
Count tests are not my favorite, although I understand at times they are necessary. If there's any way to peek in the notifications themselves and looking for a particular identifier that would be preferred.

Reply:
I'm not sure what else I'd look for. The important part is that X user gets a chat mention 🤔